mage_ai/api/resources/UserResource.py

```diff
@@ -105,20 +125,44 @@ async def _create_callback(resource):
     def update(self, payload, **kwargs):
         error = ApiError.RESOURCE_INVALID.copy()
 
-        if self.current_user.is_admin:
-            if self.owner:
-                error.update(
-                    {'message': 'Admins cannot update users who are Owners.'})
-                raise ApiError(error)
-            elif self.is_admin and self.current_user.id != self.id:
-                error.update(
-                    {'message': 'Admins cannot update users who are Admins.'})
-                raise ApiError(error)
-            elif payload.get('roles') and int(payload.get('roles')) & 1 != 0:
+        if 'roles_new' in payload:
+            roles_new = payload.get('roles_new', [])
+            roles_new = list(map(
+                lambda role_id: Role.query.get(int(role_id)),
+                roles_new,
+            ))
+            payload['roles_new'] = roles_new
```

Comment:
Do you need to throw error on None here?

Reply:
ah yeah i'll do that